app/src/main/java/net/squanchy/proximity/near/NearITProximityProvider.java

```diff
@@ -4,31 +4,47 @@
 import android.util.Log;
 
 import net.squanchy.R;
+import net.squanchy.proximity.ProximityEvent;
 import net.squanchy.proximity.ProximityProvider;
 
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.schedulers.Schedulers;
 import it.near.sdk.Geopolis.Beacons.Ranging.ProximityListener;
 import it.near.sdk.NearItManager;
+import it.near.sdk.Reactions.CustomJSON.CustomJSON;
 
-public class NearITProximityProvider implements ProximityProvider{
+public class NearITProximityProvider implements ProximityProvider {
 
-    NearItManager nearItManager;
+    private final NearItManager nearItManager;
+    private static final String PROXIMITY_ACTION = "action";
+    private static final String SUBJECT_ID = "subject";
 
-    public NearITProximityProvider(Context context) {
-        Log.d("NearITProximityProvider", "new instance created");
-        nearItManager = new NearItManager(context.getApplicationContext(), context.getString(R.string.nearit_api_key));
+    public NearITProximityProvider(NearItManager nearItManager) {
+        this.nearItManager = nearItManager;
         // TODO set a notification icon if needed
     }
 
-    public void registerForProximityEvents(ProximityListener listener) {
-        nearItManager.addProximityListener(listener);
-    }
-
-    public void deregisterForProximityEvents(ProximityListener listener) {
-        nearItManager.removeProximityListener(listener);
-    }
-
     @Override
     public void startRadar() {
         nearItManager.startRadar();
```

Comment:
it can't be in the observer creation because the radar needs to always be on to receive notification in the background (not the ones delivered by the observer, I'm talking about system notification, that's already handled by the SDK itslef).
It's in HomeActivity because it will be called only after receiving the location permission (they will probably be asked in HomeActivity but that's TBD)
And by the way, I've added the stopRadar method.